ksqldb-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadataImpl.java

```diff
@@ -105,7 +107,24 @@ public PersistentQueryMetadataImpl(
         maxQueryErrorsQueueSize,
         retryBackoffInitialMs,
         retryBackoffMaxMs,
-        listener
+        new QueryMetadata.Listener() {
+          @Override
+          public void onError(final QueryMetadata queryMetadata, final QueryError error) {
+            listener.onError(queryMetadata, error);
+            scalablePushRegistry.get().onError();
```

Comment:
Do `scalablePushRegistry.ifPresent`.  Otherwise, if SPQs are disabled, this will throw an exception.